### PR TITLE
feat: Allow changing foreground/background colors

### PR DESCRIPTION
**Pull request by Shriukan33**
---
name: Pull request
about: Submit a pull request for this project
assignees: fabiocaccamo

---

**Describe your changes**
I added 4 customization options to Themes :
* Body background color
* Foreground color
* Quiet color
* Loud color

And extended French translation for the newly added fields.
I couldn't find places where loud color was used, but it felt awkward to add the option for the quiet color without the loud color. 

**Related issue**
#291 

**Checklist before requesting a review**
- [x] I have performed a self-review of my code.
- [NA] I have added tests for the proposed changes.
- [x] I have run the tests and there are no errors.

### CODE REVIEW

**Comment by BMourguesFieldbox** [comment]
The linting keeps failing, maybe I should drop the translation commit?

It's my first time updating locale, maybe I'm doing this wrong @fabiocaccamo ?

**Comment by fabiocaccamo** [comment]
@Shriukan33 thanks for this PR, I will review it as soon as possible!

**Comment by fabiocaccamo** [comment]
@BMourguesFieldbox have you followed these steps: https://github.com/fabiocaccamo/django-admin-interface#translate-into-another-language ?

**Comment by fabiocaccamo** [comment]
This PR would probably fix #291.

**Comment by BMourguesFieldbox** [comment]
> @BMourguesFieldbox have you followed these steps? https://github.com/fabiocaccamo/django-admin-interface#translate-into-another-language

Sorry I missed it, I've gone through the proper commands and repushed ! :)

**Comment by BMourguesFieldbox** [comment]
```shell
┌──(admin-interface)─(bmourgues㉿FRL167)-[~/Documents/Github/django-admin-interface/django-admin-interface]
└─$ python -m django makemessages --ignore ".tox" --ignore "venv" --add-location "file" --extension "html,py" --locale "fr"
processing locale fr
                                                                                                                                                                     
┌──(admin-interface)─(bmourgues㉿FRL167)-[~/Documents/Github/django-admin-interface/django-admin-interface]
└─$ python -m django compilemessages --ignore ".tox" --ignore "venv"                                                       
File “/home/bmourgues/Documents/Github/django-admin-interface/django-admin-interface/admin_interface/locale/ru/LC_MESSAGES/django.po” is already compiled and up to date.
File “/home/bmourgues/Documents/Github/django-admin-interface/django-admin-interface/admin_interface/locale/de/LC_MESSAGES/django.po” is already compiled and up to date.
File “/home/bmourgues/Documents/Github/django-admin-interface/django-admin-interface/admin_interface/locale/pl/LC_MESSAGES/django.po” is already compiled and up to date.
File “/home/bmourgues/Documents/Github/django-admin-interface/django-admin-interface/admin_interface/locale/pt_BR/LC_MESSAGES/django.po” is already compiled and up to date.
File “/home/bmourgues/Documents/Github/django-admin-interface/django-admin-interface/admin_interface/locale/tr/LC_MESSAGES/django.po” is already compiled and up to date.
File “/home/bmourgues/Documents/Github/django-admin-interface/django-admin-interface/admin_interface/locale/fa/LC_MESSAGES/django.po” is already compiled and up to date.
File “/home/bmourgues/Documents/Github/django-admin-interface/django-admin-interface/admin_interface/locale/it/LC_MESSAGES/django.po” is already compiled and up to date.
File “/home/bmourgues/Documents/Github/django-admin-interface/django-admin-interface/admin_interface/locale/es/LC_MESSAGES/django.po” is already compiled and up to date.
processing file django.po in /home/bmourgues/Documents/Github/django-admin-interface/django-admin-interface/admin_interface/locale/fr/LC_MESSAGES
                                                                                                                                                                                                                                                                                                                                    
┌──(admin-interface)─(bmourgues㉿FRL167)-[~/Documents/Github/django-admin-interface/django-admin-interface]
└─$ git status --porcelain
                                                                                                                                                                     
┌──(admin-interface)─(bmourgues㉿FRL167)-[~/Documents/Github/django-admin-interface/django-admin-interface]
└─$ 
```
No diff, yet it still fails?

File: admin_interface/locale/fr/LC_MESSAGES/django.po
Comment (merwok) [inline admin_interface/locale/fr/LC_MESSAGES/django.po:183]: It doesn’t seem correct to have this more detail in the translation and not the source strings: the English version will miss the details, and the other translators too!

Comment (BMourguesFieldbox) [reply admin_interface/locale/fr/LC_MESSAGES/django.po:183]: So what's the play here? Should I add the details in the verbose name in the models.py file and then edit this one?
Sorry this is my first time contributing on langage :/

Comment (fabiocaccamo) [reply admin_interface/locale/fr/LC_MESSAGES/django.po:183]: You should just translate the original message without adding extra informations, eg. `msgstr "couleur du premier plan"`

Comment (merwok) [reply admin_interface/locale/fr/LC_MESSAGES/django.po:183]: Or add the full message in English in the source code!

Comment (BMourguesFieldbox) [reply admin_interface/locale/fr/LC_MESSAGES/django.po]: I went for an intermediate solution and instead I extended the help text. 
Instead of adding details in the verbose name, I felt that these details belong more to the help text.

Also added the translation for those new help texts.

**Comment by BMourguesFieldbox** [comment]
@fabiocaccamo Just a reminder if you have some time to pull the trigger on the CI :)

**Comment by Shriukan33** [comment]
Hey, just a heads up for this PR :)

**Comment by BMourguesFieldbox** [comment]
> @Shriukan33 sorry for the delay in doing the review, but I'm quite busy these days and I'm not able to find the time. I will do it as soon as possible, this has highest-priority in my [open-source todo list](https://github.com/users/fabiocaccamo/projects/6/views/1).

No problem, I understand ! Life gets in the way sometimes, good luck in whatever keeps you busy :)

**Comment by selected-pixel-jameson** [comment]
@fabiocaccamo Is there something blocking this from being merged? Would be awesome if we could get this in there. This project has everything I'm looking for except this feature.

**Comment by fabiocaccamo** [comment]
@selected-pixel-jameson I haven't had time to review and test this PR yet.

**Comment by BMourguesFieldbox** [comment]
> @fabiocaccamo Is there something blocking this from being merged? Would be awesome if we could get this in there. This project has everything I'm looking for except this feature.

In the meantime, you could use my fork as source in your requirements like so : 

```
django==5.*
git+https://github.com/Shriukan33/django-admin-interface.git@Allow-changing-background-color
```
Note that you need to have git installed on the image if you're using docker.

**Comment by selected-pixel-jameson** [comment]
@BMourguesFieldbox I'll take a look, but I'm kind of having concerns even moving forward with this solution as it sounds like the maintainer is potentially running into time constraint issues, which is understandable, I just don't want to implement a solution that in the end is going to run into maintenance issues.

**Comment by fabiocaccamo** [comment]
@BMourguesFieldbox @Shriukan33 I'll try to review this as soon as possible!

**Comment by BMourguesFieldbox** [comment]
> @BMourguesFieldbox I'll take a look, but I'm kind of having concerns even moving forward with this solution as it sounds like the maintainer is potentially running into time constraint issues, which is understandable, I just don't want to implement a solution that in the end is going to run into maintenance issues.

Understandable, this lib doesn't do much aside from offering a way to edit the admin templates, so it doesn't really relate to app's features, I'd say it's quite low risk (unless you have lots of features in the admin) 



> @BMourguesFieldbox @Shriukan33 @selected-pixel-jameson I'll try to review this as soon as possible!

Thanks ! :D

**Comment by BMourguesFieldbox** [comment]
Hey, just a heads up I rebased on current main, as it had like 50 new commits behind :)

File: admin_interface/models.py
Comment (fabiocaccamo) [inline admin_interface/models.py:183]: Could you improve this `help_text` too please?

File: admin_interface/locale/it/LC_MESSAGES/django.po
Comment (fabiocaccamo) [inline admin_interface/locale/it/LC_MESSAGES/django.po:190]: This is wrong, please leave it empty.

File: admin_interface/locale/it/LC_MESSAGES/django.po
Comment (fabiocaccamo) [inline admin_interface/locale/it/LC_MESSAGES/django.po:183]: Please do not include the default color in the translated string (apply this change to all similar occurrences).

File: admin_interface/locale/it/LC_MESSAGES/django.po
Comment (fabiocaccamo) [inline admin_interface/locale/it/LC_MESSAGES/django.po:204]: This is wrong, leave it empty please.

File: admin_interface/locale/it/LC_MESSAGES/django.po
Comment (fabiocaccamo) [inline admin_interface/locale/it/LC_MESSAGES/django.po:210]: This is wrong, leave it empty please.

**Review by fabiocaccamo — CHANGES_REQUESTED** [review]

File: admin_interface/locale/it/LC_MESSAGES/django.po
Comment (merwok) [reply admin_interface/locale/it/LC_MESSAGES/django.po:190]: it is a fuzzy entry, so won’t be displayed at run-time

(this happens when strings extraction finds a matching existing entry for a string that’s not the same, the previous or similar msgstr is added as a hint for translators but the entry is marked fuzzy)

File: admin_interface/models.py
Comment (BMourguesFieldbox) [reply admin_interface/models.py:183]: I couldn't find where loud color was used, 

> I couldn't find places where loud color was used, but it felt awkward to add the option for the quiet color without the loud color.

File: admin_interface/locale/it/LC_MESSAGES/django.po
Comment (BMourguesFieldbox) [reply admin_interface/locale/it/LC_MESSAGES/django.po:190]: So nothing to delete in the end?

**Comment by Shriukan33** [comment]
Removed the default values in the help text

Left the fuzzy occurrences, as they are auto generated and not displayed at runtime

**Comment by BMourguesFieldbox** [comment]
Thanks!

Le ven. 13 sept. 2024, 15:23, Fabio Caccamo ***@***.***> a
écrit :

**Comment by fabiocaccamo** [comment]
@BMourguesFieldbox sorry, I reverted the merging because it didn't worked well.

**Comment by Shriukan33** [comment]
Ha, what happened? I use it on production and it doesn't seem to be problematic?

Envoyé à partir de Outlook pour Android<https://aka.ms/AAb9ysg>
________________________________

**Pull request merged**
